android/libraries/rib-base/src/main/java/com/badoo/ribs/android/recyclerview/Adapter.kt

```diff
@@ -0,0 +1,101 @@
+package com.badoo.ribs.android.recyclerview
+
+import android.os.Parcelable
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.recyclerview.widget.RecyclerView
+import com.badoo.ribs.android.recyclerview.RecyclerViewHost.HostingStrategy.EAGER
+import com.badoo.ribs.android.recyclerview.RecyclerViewHost.HostingStrategy.LAZY
+import com.badoo.ribs.android.recyclerview.RecyclerViewHost.Input
+import com.badoo.ribs.android.recyclerview.RecyclerViewHostFeature.State.Entry
+import com.badoo.ribs.android.recyclerview.RecyclerViewHostRouter.Configuration.Content.Item
+import com.badoo.ribs.core.routing.configuration.ConfigurationKey
+import io.reactivex.functions.Consumer
+import java.util.UUID
+
+internal class Adapter<T : Parcelable>(
+    private val hostingStrategy: RecyclerViewHost.HostingStrategy,
+    initialEntries: List<Entry<T>>? = null,
+    private val router: RecyclerViewHostRouter<T>,
+    private val viewHolderLayoutParams: FrameLayout.LayoutParams
+) : RecyclerView.Adapter<Adapter.ViewHolder>(),
+    Consumer<RecyclerViewHostFeature.State<T>> {
+
+    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        var configurationKey: ConfigurationKey? = null
+        var uuid: UUID? = null
+    }
+
+    private var items: List<Entry<T>> = initialEntries ?: emptyList()
+
+    override fun getItemCount(): Int =
+        items.size
+
+    override fun accept(state: RecyclerViewHostFeature.State<T>) {
+        items = state.items
+
+        when (state.lastCommand) {
+            null -> { /* No-op when restored from TimeCapsule or genuinely empty state */ }
+            is Input.Add -> {
+                eagerAdd(state.items.last())
```

Comment:
What if we want to add N elements, or remove, or replace. Can we add more operations here?